examples/ota-provider-app/ota-provider-common/OTAProviderExample.cpp

```diff
@@ -178,62 +248,33 @@ EmberAfStatus OTAProviderExample::HandleQueryImage(chip::app::CommandHandler * c
+        // Initialize the transfer session in prepartion for a BDX transfer
+        mBdxOtaSender.SetFilepath(otaFilePath);
+        BitFlags<TransferControlFlags> bdxFlags;
+        bdxFlags.Set(TransferControlFlags::kReceiverDrive);
+        CHIP_ERROR err = mBdxOtaSender.PrepareForTransfer(&chip::DeviceLayer::SystemLayer(), chip::bdx::TransferRole::kSender,
+                                                          bdxFlags, kMaxBdxBlockSize, kBdxTimeout, kBdxPollFreq);
+        if (err != CHIP_NO_ERROR)
         {
-            queryStatus = OTAQueryStatus::kUpdateAvailable;
-
-            // Initialize the transfer session in prepartion for a BDX transfer
-            mBdxOtaSender.SetFilepath(otaFilePath);
-            BitFlags<TransferControlFlags> bdxFlags;
-            bdxFlags.Set(TransferControlFlags::kReceiverDrive);
-            CHIP_ERROR err = mBdxOtaSender.PrepareForTransfer(&chip::DeviceLayer::SystemLayer(), chip::bdx::TransferRole::kSender,
-                                                              bdxFlags, kMaxBdxBlockSize, kBdxTimeout, kBdxPollFreq);
-            if (err != CHIP_NO_ERROR)
-            {
-                ChipLogError(BDX, "Failed to initialize BDX transfer session: %s", chip::ErrorStr(err));
-                return EMBER_ZCL_STATUS_FAILURE;
-            }
+            ChipLogError(BDX, "Failed to initialize BDX transfer session: %s", chip::ErrorStr(err));
+            return EMBER_ZCL_STATUS_FAILURE;
         }
-        else
-        {
-            queryStatus = OTAQueryStatus::kNotAvailable;
-            ChipLogError(SoftwareUpdate, "No OTA file configured on the Provider");
-        }
-        break;
-    }
-    case kRespondWithBusy: {
-        queryStatus = OTAQueryStatus::kBusy;
-        break;
-    }
-    case kRespondWithNotAvailable: {
-        queryStatus = OTAQueryStatus::kNotAvailable;
-        break;
-    }
-    default:
-        queryStatus = OTAQueryStatus::kNotAvailable;
-    }
 
-    QueryImageResponse::Type response;
-    response.status = queryStatus;
-    response.delayedActionTime.Emplace(mDelayedActionTimeSec);
-    if (queryStatus == OTAQueryStatus::kUpdateAvailable)
-    {
         response.imageURI.Emplace(chip::CharSpan::fromCharString(uriBuf));
         response.softwareVersion.Emplace(newSoftwareVersion);
         response.softwareVersionString.Emplace(chip::CharSpan::fromCharString(newSoftwareVersionString));
         response.updateToken.Emplace(chip::ByteSpan(updateToken));
-        response.userConsentNeeded.Emplace(userConsentNeeded);
-        // Could also just not send metadataForRequestor at all.
-        response.metadataForRequestor.Emplace(chip::ByteSpan());
     }
+
+    response.status = queryStatus;
+    response.delayedActionTime.Emplace(delayedActionTimeSec);
+    response.userConsentNeeded.Emplace(requestorCanConsent);
+    // Could also just not send metadataForRequestor at all.
+    response.metadataForRequestor.Emplace(chip::ByteSpan());
```

Comment:
Maybe for test coverage, send empty metadata when client `(nodeId % 2) == 0` and don't send it otherwise. That way, deterministically, half the clients will get empty metadata, half not. That would help test the different handling of metadata, compared to a single policy. Can be done as a follow-up.

@carol-apple What do you think ^^^

Reply:
Let's do that in this PR